### PR TITLE
Tidy src/atoms/second_order_cone

### PR DESCRIPTION
**Pull request by odow**
x-ref #538

### CODE REVIEW

File: src/atoms/second_order_cone/rationalnorm.jl [hunk 1]
```diff
@@ -60,7 +50,8 @@ function rationalnorm(x::AbstractExpr, k::Rational{Int})
         if row == 1 || col == 1
             return RationalNormAtom(abs(x), k)
         end
-    else
-        return RationalNormAtom(x, k)
+        # FIXME(odow): what happens if x is a matrix?
```
Comment (odow) [inline src/atoms/second_order_cone/rationalnorm.jl:53]: Follow-up later. The function previously returned `nothing` if `x` was a complex matrix.

File: src/atoms/second_order_cone/quadform.jl [hunk 2]
```diff
@@ -39,33 +41,29 @@ function is_psd(
...
     if length(size(A)) != 2 || size(A, 1) != size(A, 2)
         error("Quadratic form only takes square matrices")
-    end
-    if !LinearAlgebra.ishermitian(A)
+    elseif !LinearAlgebra.ishermitian(A)
         error("Quadratic form only defined for Hermitian matrices")
     end
-    if assume_psd
-        factor = 1
+    factor = if assume_psd || is_psd(A)
```
Comment (odow) [inline src/atoms/second_order_cone/quadform.jl:61]:
```suggestion
    factor = if assume_psd || is_psd(A)
        1
```

File: src/atoms/second_order_cone/rationalnorm.jl [hunk 3]
```diff
@@ -9,36 +6,29 @@
...
 mutable struct RationalNormAtom <: AbstractExpr
     children::Tuple{AbstractExpr}
     size::Tuple{Int,Int}
     k::Rational{Int64}
 
     function RationalNormAtom(x::AbstractExpr, k::Rational{Int})
-        children = (x,)
-        k >= 1 || error("p-norms not defined for p < 1")
-        return new(children, (1, 1), k)
+        if k >= 1
```
Comment (odow) [inline src/atoms/second_order_cone/rationalnorm.jl:16]:
```suggestion
        if k < 1
```

File: src/atoms/second_order_cone/norm2.jl [hunk 4]
```diff
@@ -1,59 +1,32 @@
...
-    return ConvexVexity()
-end
+curvature(::EucNormAtom) = ConvexVexity()
 
-function evaluate(x::EucNormAtom)
-    return norm(vec(evaluate(x.children[1])))
-end
+evaluate(x::EucNormAtom) = norm(vec(evaluate(x.children[1])))
 
-## Create a new variable euc_norm to represent the norm
```
Comment (blegat) [inline src/atoms/second_order_cone/norm2.jl:36]: Maybe move it to a docstring ?

Comment (odow) [reply src/atoms/second_order_cone/norm2.jl:36]: This comment was outdated so I deleted it. The function did not in fact create a new variable `euc_norm` to represent the norm. I think it mean `t`, and okay, by reading the function I see it is building `[t, x] in SOC`. I don't think it needs a comment saying that.

**Pull request merged**